sdks/rust/src/sdk.rs

```diff
@@ -12,349 +12,211 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::env;
-use std::sync::{Arc, Mutex};
-use std::thread::sleep;
-use std::time::Duration;
+use std::{env, time::Duration};
+use tonic::transport::Channel;
 
-use futures::future::TryFutureExt;
-use futures::prelude::*;
-use futures::FutureExt;
-use futures01::future::Future;
+mod api {
+    tonic::include_proto!("agones.dev.sdk");
+}
+
+use api::sdk_client::SdkClient;
+pub use api::GameServer;
 
-use grpcio;
-use protobuf::Message;
+pub type WatchStream = tonic::Streaming<GameServer>;
 
-use super::alpha::*;
-use super::errors::*;
-use super::grpc::sdk;
-use super::grpc::sdk_grpc;
-use super::types::*;
+use crate::{
+    alpha::Alpha,
+    errors::{Result},
+};
+
+#[inline]
+fn empty() -> api::Empty {
+    api::Empty {}
+}
 
 /// SDK is an instance of the Agones SDK
+#[derive(Clone)]
 pub struct Sdk {
-    client: Arc<sdk_grpc::SdkClient>,
-    health: Arc<Mutex<Option<grpcio::ClientCStreamSender<sdk::Empty>>>>,
-    health_receiver: Arc<Mutex<Option<grpcio::ClientCStreamReceiver<sdk::Empty>>>>,
-    alpha: Arc<Alpha>,
+    client: SdkClient<Channel>,
+    alpha: Alpha,
```

Comment:
Arc is not needed for the tonic clients, they are internally cloneable, the health stream is no longer needed here, which I'll comment on below where it was changed.